File: gost-34-10-94/CryptoLR5_kotlin/src/LinearCongruentialGenerator.kt
class LinearCongruentialGenerator(var b: Long, var seed: Long, var c: Long, var m: Long) {

    fun getRandValue(): Long {
        val new: Long = (b * seed + c).mod(m)
        seed = new
        return new
    }

    fun getRandValues(n: Int): Array<Long> {
        return Array(n) { getRandValue() }
    }
}

File: gost-34-10-94/CryptoLR5_cxx/LinearCongruentialGenerator.cpp
#include "LinearCongruentialGenerator.h"
#include "Utils.h"

LinearCongruentialGenerator::LinearCongruentialGenerator(long long b, long long seed, long long c, long long m) {
    this->b = b;
    this->seed = seed;
    this->c = c;
    this->m = m;
}

long long LinearCongruentialGenerator::getRandValue() {
    long long value = mod(b * seed + c, m);
    seed = value;
    return value;
}
